lib/Telemetry/src/DispatcherFactory.h

```diff
@@ -0,0 +1,114 @@
+#include "DataQueue.h"
+#include "LogCommand.h"
+#include "Command.h"
+#include "Dispatcher.h"
+#include "Particle.h"
+#include "IntervalLogger.h"
+
+#define MAX_NUM_INTERVALS 5
+
+#ifndef _DISPTACHER_FACTORY_H_
+#define _DISPATCHER_FACTORY_H_
+
+// This isn't really a factory but it has a similar purpose:
+// Add all the commands you want to this and call build().  It
+// produces a dispatcher with all logging functionality already set up
+// You can delete this safely after build() has been called
+class DispatcherFactory {
+    public:
+        /**
+         * Constructs new DispatcherFactory with number of commands and data queue
+         * 
+         * @param numCommands The number of sensor functions you will add to the factory
+         * Note: this must be accurate and has potential to cause a segfault if the value is too low
+         * @param dataQ the Data Queue from main
+         * */
+        DispatcherFactory(uint16_t numCommands, DataQueue *dataQ) {
+            _dataQ = dataQ;
+            _commands = new Command*[numCommands];
+            _intervals = new uint16_t[MAX_NUM_INTERVALS] ();
+            _numCommandsOnIntervals = new uint16_t[MAX_NUM_INTERVALS] ();
+            _numCommands = 0;
+            _numIntervals = 0;
+        }
+        
+        /**
+         * Destructor which frees unused heap data (can safely be called after build)
+        **/
+        ~DispatcherFactory() {
+            delete _intervals;
+            delete _numCommandsOnIntervals;
+        }
+
+        /**
+         * Adds a Sensor-property getter function, the property name, and interval at which to call it
+         * 
+         * @param S the address of the Sensor whose getter you want to be called
+         * @param propertyName the publish-name of the property which the getter returns
+         * @param func function pointer to Sensor member function with return type of R
+         * @param interval the interval (in seconds) at which you want to log data from Sensor's getter
+         **/
+        template <class S, class R>
+        void add(S *sensor, String propertyName, R (S::*func)(), uint16_t interval) {
+            _commands[_numCommands++] = new LogCommand<S, R>(sensor, propertyName, func, interval);
```

Comment:
I don't see where these new LogCommands are deleted. I'm assuming they get passed to new intervalLoggers and continue to be used. These would create a memory leak if you call the destructor before calling build(). Instead of having a destructor, why not make all the delete calls at the end of build since this object wouldn't be used any more?

Reply:
they're actually being used when log() is called on the IntervalLoggers, so they need to stay in memory

Reply:
That makes sense. I was just thinking about the case where you call the destructor before calling build(), you'd _technically_ have a memory leak. But that's not the case with how this class is meant to be used, so I think it's probably fine as is.

Reply:
Interesting.  Calling delete before build would cause memory leaks in other ways because the destructor won't delete any of the Commands either, just the int and pointer arrays (I've changed the destructor so that it calls delete[] on all the dynamically allocated member arrays).

I feel like this is ok because like you said the factory doesn't serve a purpose beyond building a dispatcher, so the chance that somebody would create and destroy one without calling build are pretty slim